src/sentry/static/sentry/app/views/projectKeyDetails.jsx

```diff
@@ -2,43 +2,179 @@ import React from 'react';
 import DocumentTitle from 'react-document-title';
 import underscore from 'underscore';
 import {browserHistory} from 'react-router';
+import idx from 'idx';
 
 import ApiMixin from '../mixins/apiMixin';
 import AutoSelectText from '../components/autoSelectText';
 import DateTime from '../components/dateTime';
+import HookStore from '../stores/hookStore';
 import IndicatorStore from '../stores/indicatorStore';
 import LoadingError from '../components/loadingError';
 import LoadingIndicator from '../components/loadingIndicator';
-import OrganizationState from '../mixins/organizationState';
-import {BooleanField, FormState, TextField} from '../components/forms';
+import ProjectState from '../mixins/projectState';
+import StackedBarChart from '../components/stackedBarChart';
+import {
+  BooleanField,
+  FormState,
+  NumberField,
+  Select2Field,
+  TextField
+} from '../components/forms';
 import {t, tct} from '../locale';
 
+const KeyStats = React.createClass({
+  mixins: [ApiMixin],
+
+  getInitialState() {
+    let until = Math.floor(new Date().getTime() / 1000);
+    let since = until - 3600 * 24 * 30;
+
+    return {
+      since: since,
+      until: until,
+      loading: true,
+      error: false,
+      stats: null,
+      emptyStats: false
+    };
+  },
+
+  componentWillMount() {
+    this.fetchData();
+  },
+
+  fetchData() {
+    let {keyId, orgId, projectId} = this.props.params;
+    this.api.request(`/projects/${orgId}/${projectId}/keys/${keyId}/stats/`, {
+      query: {
+        since: this.state.since,
+        until: this.state.until,
+        resolution: '1d'
+      },
+      success: data => {
+        let emptyStats = true;
+        let stats = data.map(p => {
+          if (p.total) emptyStats = false;
+          return {
+            x: p.ts,
+            y: [p.accepted, p.dropped, p.filtered]
+          };
+        });
+        this.setState({
+          stats: stats,
+          emptyStats: emptyStats,
+          error: false,
+          loading: false
+        });
+      },
+      error: () => {
+        this.setState({error: true, loading: false});
+      }
+    });
+  },
+
+  renderTooltip(point, pointIdx, chart) {
+    let timeLabel = chart.getTimeLabel(point);
+    let [accepted, dropped, filtered] = point.y;
+
+    let value = `${accepted.toLocaleString()} accepted`;
+    if (dropped) {
+      value += `<br>${dropped.toLocaleString()} rate limited`;
+    }
+    if (filtered) {
+      value += `<br>${filtered.toLocaleString()} filtered`;
+    }
+
+    return (
+      '<div style="width:150px">' +
+      `<div class="time-label">${timeLabel}</div>` +
+      `<div class="value-label">${value}</div>` +
```

Comment:
Potential XSS vulnerability here (the tooltip HTML won't be escaped). Worth just leaving a comment that these values are statically defined so there's "no risk", or just wrapping each of these in `_.escape` for good measure.